### PR TITLE
add remaining math methods (#524)

### LINKED ISSUE #524: Implement the rest of `Math` methods
Implement `Math` methods:
 - [x] `Math.clz32()`
 - [x] `Math.expm1()`
 - [x] `Math.fround()`
 - [x] `Math.hypot()`
 - [x] `Math.log1p()`
 - [x] `Math.imul()`

### PR DESCRIPTION
**Pull request by mr-rodgers**
This Pull Request fixes/closes #524.

It changes the following (work-in-progress):

 - [x] Math.clz32()
 - [x] Math.expm1()
 - [x] Math.fround()
 - [x] Math.hypot()
 - [x] Math.log1p()
 - [x] Math.imul()

### CODE REVIEW

File: boa/src/builtins/math/mod.rs [hunk 1]
```diff
@@ -165,6 +166,25 @@ impl Math {
...
+    /// [spec]: https://tc39.es/ecma262/#sec-math.clz32
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/clz32
+    pub(crate) fn clz32(_: &Value, args: &[Value], _: &mut Interpreter) -> ResultValue {
+        Ok(args
+            .get(0)
+            .and_then(|x| f64::from(x).to_u32())
+            .map_or(32usize, |u| {
+                let s = format!("{:032b}", u);
+                s.chars().take_while(|x| *x == '0').count()
+            })
```
Comment (HalidOdat) [inline boa/src/builtins/math/mod.rs:184]:
```suggestion
            .and_then(|x| i32::from(x).leading_zeros())
            .unwrap_or(32)
```
A better way of doing this would we to use [`.leading_zeros()`](https://doc.rust-lang.org/std/primitive.i32.html#method.leading_zeros) which would eliminate the allocation from `format!`

File: boa/src/builtins/math/tests.rs [hunk 2]
```diff
@@ -285,6 +435,36 @@ fn log() {
     assert_eq!(c, String::from("NaN"));
 }
 
+#[test]
+fn log1p() {
+    let realm = Realm::create();
+    let mut engine = Interpreter::new(realm);
+    let init = r#"
+        var a = Math.log1p(1);
```
Comment (pedropaulosuzuki) [inline boa/src/builtins/math/tests.rs:443]: Maybe we should add tests for empty as well.
```js
const g = Math.log1p();
```

File: boa/src/builtins/math/tests.rs [hunk 3]
```diff
@@ -264,6 +330,90 @@ fn floor() {
...
+
+#[test]
+fn imul() {
+    let realm = Realm::create();
+    let mut engine = Interpreter::new(realm);
+    let init = r#"
+        var a = Math.imul(3, 4);
+        var b = Math.imul(-5, 12);
+        var c = Math.imul(0xffffffff, 5);
+        var d = Math.imul(0xfffffffe, 5);
```
Comment (pedropaulosuzuki) [inline boa/src/builtins/math/tests.rs:401]: Maybe we should add tests for empty as well.
```js
const e = Math.imul();
const f = Math.imul(30);
```

File: boa/src/builtins/math/tests.rs [hunk 4]
```diff
@@ -264,6 +330,90 @@ fn floor() {
...
+    assert_eq!(f.to_number(), -5.050_000_190_734_863);
+}
+
+#[test]
+fn hypot() {
+    let realm = Realm::create();
+    let mut engine = Interpreter::new(realm);
+    let init = r#"
+        var a = Math.hypot();
+        var b = Math.hypot(3, 4);
```
Comment (pedropaulosuzuki) [inline boa/src/builtins/math/tests.rs:369]: Maybe we should add tests for single element as well.
```js
const g = Math.hypot(9);
```

File: boa/src/builtins/math/tests.rs [hunk 5]
```diff
@@ -264,6 +330,90 @@ fn floor() {
     assert_eq!(c.to_number(), 3_f64);
 }
 
+#[test]
+fn fround() {
+    let realm = Realm::create();
+    let mut engine = Interpreter::new(realm);
+    let init = r#"
+        var a = Math.fround(NaN);
```
Comment (pedropaulosuzuki) [inline boa/src/builtins/math/tests.rs:338]: Maybe we should add tests for empty as well.
```js
const g = Math.fround();
```
